### ISSUE TITLE
Accessibility: reflect dynamic changes of label elements

### ISSUE THREAD

**Issue by elenastoyanovaa** [SAP/ui5-webcomponents#6315]
### **Description**
When a user links a textual element (it could be a ui5-label, title or even a simple span) via 'accessible-name-ref' to a web components and the text of this elements changes, the changes should be applied to the aria-label of the labeled element also.
This issue should cover the scenario when an element and a ui5-label are linked with the 'for' attribute as well

### **Isolated Example**
https://codesandbox.io/s/ui5-webcomponents-forked-w66sxu?fontsize=14&hidenavigation=1&theme=dark

- Change the text of the labels.
- Inspect the inputs - the aria-label value is still the previous label textual content.

### **Proposed Solution**
The framework class `AriaLabelHelper` needs to be enhanced and use MutationObserver to reflect runtime changes of the referenced DOM elements.

### **Context**
- UI5 Web Components version: all
- OS/Platform: n/a
- Browser: n/a
- Affected component: ui5-label, all components implementing accessible-name-ref

### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High
